plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/swarm/client/SwarmDockerConnector.java

```diff
@@ -112,6 +121,33 @@ public ContainerCreated createContainer(CreateContainerParams params) throws IOE
         }
     }
 
+    @Override
+    public void startExec(StartExecParams params, @Nullable MessageProcessor<LogMessage> execOutputProcessor)
+            throws IOException {
+        try {
+            super.startExec(params, execOutputProcessor);
+        } catch (ExecNotFoundException e) {
+            // Sometimes swarm returns this error for unknown reason, see https://github.com/docker/swarm/issues/2664
+            // Log additional info to find out if this endpoint knows about exec at exactly that time
+            logMissingExecInfo(params.getExecId());
+            try {
+                // Wait in case swarm needs some time to find exec
+                Thread.sleep(1000);
```

Comment:
I'm afraid that 1 second could be not enough. Please set a bit longer delay (maybe 5 seconds is long delay, but it will be better to wait it (almost for sure that exec will be visible by swarm) than give workspace starting error to user)